### PR TITLE
[Merged by Bors] - fix: Operator errors out when credentialsSecret is missing

### PR DESCRIPTION
**Pull request by Maleware**
# Description

Fix for logging an error if secrets are not found

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/operator-binary/src/airflow_db_controller.rs [hunk 1]
```diff
@@ -102,6 +102,8 @@ pub enum Error {
         source: crate::product_logging::Error,
         cm_name: String,
     },
+    #[snafu(display("unable to find SecretClass with airflow credentials"))]
+    SecretNotFound,
```
Comment (maltesander) [inline rust/operator-binary/src/airflow_db_controller.rs:106]: Can we mention the credential secrets name and something like "spec.credentialsSecret" in the error?

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,7 @@
 - Fix LDAP tests for Openshift ([#270]).
 - Missing CRD defaults for `status.conditions` field ([#277]).
 - Support Airflow `2.6.1` ([#284]).
+- Operator errors out when credentialsSecret is missing ([#293])
```
Comment (maltesander) [inline CHANGELOG.md:11]: For consistency:
```suggestion
- Operator errors out when credentialsSecret is missing ([#293]).
```

File: rust/operator-binary/src/airflow_db_controller.rs [hunk 3]
```diff
@@ -202,6 +204,8 @@ pub async fn reconcile_airflow_db(airflow_db: Arc<AirflowDB>, ctx: Arc<Ctx>) ->
                         )
                         .await
                         .context(ApplyStatusSnafu)?;
+                } else {
+                    return SecretNotFoundSnafu.fail();
```
Comment (maltesander) [inline rust/operator-binary/src/airflow_db_controller.rs:208]: There is a `client.get_opt(secret...)` call. I would not use `get_opt` but just `get` and error out there (not tested)?

**Review by maltesander — CHANGES_REQUESTED** [review]
Minor things..

File: rust/operator-binary/src/airflow_db_controller.rs (hunk 1 above)
Comment (Maleware) [reply rust/operator-binary/src/airflow_db_controller.rs:106]: Not sure, there is no secret and therefore no name I guess. Could mention the name of the secret in the docs.

Comment (maltesander) [reply rust/operator-binary/src/airflow_db_controller.rs:106]: The credentialsSecret is a String in the CRD and is there. It just does not exist in the API server.

File: rust/operator-binary/src/airflow_db_controller.rs (hunk 3 above)
Comment (Maleware) [reply rust/operator-binary/src/airflow_db_controller.rs]: done

**Review by sbernauer — APPROVED** [review]
LGTM when tests pass

**Review by maltesander — APPROVED** [review]
LGTM!

**Comment by Maleware** [comment]
all tests passed: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/airflow-operator-it-custom/88/console

**Pull request closed without merging**